addon/utils/bind-data-test-attributes.js

```diff
@@ -15,14 +15,16 @@ export default function bindDataTestAttributes(component) {
     return;
   }
 
-  let tagName = component.get('tagName');
+  if (!component.get('supportsDataTestProperties')) {
+    let tagName = component.get('tagName');
 
-  let message = `ember-test-selectors could not bind data-test-* properties on ${component} ` +
-    `automatically because tagName is empty.`;
+    let message = `ember-test-selectors could not bind data-test-* properties on ${component} ` +
+      `automatically because tagName is empty.`;
```

Comment:
might make sense to add a link here to the new README section. something like "if this was intentional have a look at xyz to see how to avoid this assertion"